src/main/java/org/sonar/squidbridge/ProgressReport.java

```diff
@@ -74,9 +73,9 @@ public void run() {
     }
   }
 
-  public synchronized void start(Collection<File> files) {
-    count = files.size();
-    it = files.iterator();
+  public synchronized void start(Iterable<InputFile> inputFiles) {
```

Comment:
I really don't know for this and would like an advice @pynicolas .

It seems to simplify a lot things for many plugin (not for SonarJava). But it seems stupid to me to have to rely on `InputFile` (which may add external constraints to users of progressReport), while we only need them to display a path in a log. 

What do you think of only relying on an `Iterable<String> filenames` instead?